src/createRollupConfig.ts

```diff
@@ -213,6 +213,6 @@ export async function createRollupConfig(
           toplevel: opts.format === 'cjs',
           warnings: true,
         }),
-    ],
+    ].filter(Boolean) as Plugin[],
```

Comment:
This is a simple one and doesn't change any behavior, but the type-cast seemed to have eroded some typing requirements that exist without it. E.g. Rollup requires each `Plugin` to have a `name` property, and some of these custom ones here do not.

Reply:
Ahh think I need a type predicate here per https://github.com/microsoft/TypeScript/issues/20812. I've been using these type-casts after `filter` like #54 did, but a type predicate / type guard is the right way to do it that I believe will prevent eroding of types

Reply:
So I tried the below:
```typescript
].filter((e): e is Plugin => Boolean(e))
```

but that didn't fully work either. It also eroded some types, like the `name` property requirement mentioned above 😕 

With the `TSDXOptions` cast it didn't work because it doesn't seem to be able to infer the `AtLeastOne` portion 😕 

For this Rollup config piece, I was able to rewrite the `condition && plugin` to `...(condition ? [plugin] : [])` which I've done before elsewhere, and then that worked, but it is quite verbose... maybe I can abstract that into a function...